frontend/src/metabase-lib/expression.ts

```diff
@@ -25,10 +25,9 @@ export function expressionName(clause: ExpressionClause): string {
   return ML.expression_name(clause);
 }
 
-export function withExpressionName(
-  clause: ExpressionClause,
-  newName: string,
-): ExpressionClause {
+export function withExpressionName<
+  Clause extends AggregationClause | ExpressionClause,
+>(clause: Clause, newName: string): Clause {
```

Comment:
I could not make it work [with overloads](https://github.com/metabase/metabase/pull/36031/commits/f18cd7530b00c74a4881ca230dc9872ac80108ba), TS would [complain when calling the function in `AggregationPicker`](https://github.com/metabase/metabase/assets/6830683/454310f9-cba0-4075-8b7f-c4ff26b96207).